packages/pwa-kit-react-sdk/src/ssr/universal/hooks/index.js

```diff
@@ -1,14 +1,44 @@
 /*
- * Copyright (c) 2022, salesforce.com, inc.
+ * Copyright (c) 2022, Salesforce, Inc.
  * All rights reserved.
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-/* istanbul ignore file */
 
 import {useContext} from 'react'
-import {ExpressContext} from '../contexts'
+import {IsPrePassContext, ServerContext} from '../contexts'
 
-export const useExpress = () => {
-    return useContext(ExpressContext)
+/**
+ * Server context
+ * @typedef {Object} ServerContext
+ * @property {Object} req - Request object
+ * @property {Object} res - Response object
+ */
+
+/**
+ * @callback serverCtxCallback
+ * @param {ServerContext} serverContext
+ */
+
+/**
+ * Execute the given function within the server context of a server-rendered page
+ *
+ * @param {serverCtxCallback} fn - Function will be called when the server has received the request but the response is not sent yet
+ * @example
+ * useServerContext(({req, res}) => { res.status(404) })
+ */
+export const useServerContext = (fn) => {
```

Comment:
We talked about naming this hook. How it's not a good idea to have 'context', 'callback', or 'effect' in the name because it reminds you of the built-in React hooks, and our hook does not behave in a similar way to them. 

So how about `useRunOnServer`? Inspired by the name of this other hook called `useRunOnce`: https://dev.to/perssondennis/react-hook-userunonce-48a8

Reply:
My final suggestion: `useContextOnServer`

![Safari 2022-09-15 at 14 13 04](https://user-images.githubusercontent.com/847300/190510040-f6fae96e-68fa-420c-9c13-a50def81a551.png)